src/main.rs

```diff
@@ -50,13 +50,65 @@ use std::os::unix::fs::PermissionsExt;
 use std::os::unix::net::{UnixListener, UnixStream};
 use std::path::Path;
 use std::thread;
+use std::time::Duration;
 
 use anyhow::{Context, Result};
 use slog::{debug, error, o, Drain};
 
 mod ffi;
 mod handlers;
 mod protocol;
+mod semaphore;
+
+const SOCKET_PATH: &str = "/var/run/nscd/socket";
+const MAX_THREADS: usize = 64;
+const MAX_THREADS_WAIT_TIMEOUT: Duration = Duration::from_secs(5);
+
+fn main() -> Result<()> {
```

Comment:
I just noticed I reordered `main` while moving `Semaphore` to its own module. I can leave it or revert that.